src/csharp/src/ZooKeeperNetEx/zookeeper/ZooKeeper.cs

```diff
@@ -1169,5 +1169,24 @@ private static async Task<T> TryOperation<T>(ZooKeeper zk, Func<ZooKeeper, Task<
         }
 
         #endregion
+        
+        /// <summary>
+        /// Dispose of the underlying client connection.
+        /// </summary>
+        /// <param name="disposingDeterministically">Indicates whether this method is called
+        /// deterministically (i.e., from the IDisposable.Dispose method), or from the finalizer.</param>
+        protected virtual void Dispose(bool disposingDeterministically)
+        {
+            closeAsync().GetAwaiter().GetResult();
```

Comment:
When calling GetAwaiter().GetResult(), it is not guaranteed to be blocking.

The behavior of GetResult() depends on the implementation of the awaiter. While it is common for awaiters to provide a blocking behavior for GetResult(), it is not required.